### PR TITLE
Replaces the Aggregated API Server with the CustomResourceDefinitions (CRDs) solution

### PR DESCRIPTION
**Pull request by mszostok**
This PR is a 
 - [x] Feature Implementation
 - [ ] Bug Fix
 - [ ] Documentation

**What this PR does / why we need it**:

This PR replaces the Aggregated API Server with the CustomResourceDefinitions (CRDs) solution.

#### How changes were tested
- `make format`, `make verify`, `make test-unit` were executed and passed with success
- The [E2E tests](https://github.com/kubernetes-incubator/service-catalog/tree/v0.2.0/test/e2e) from the **v0.2.0** tag were executed and all 5 scenarios passed with success
- The [E2E tests](https://github.com/kyma-project/kyma/tree/master/tests/acceptance/servicecatalog) from our Kyma platform were executed and passed with success.
- Manual tests were executed against the minikube and GKE cluster installation.
  Scenarios:
  - creating/deleting cluster and namespace scoped brokers (with basic auth, with bearer auth, without auth)
  - provisioning/deprovisioning instances from cluster and namespace scoped brokers
  - binding/unbinding

All of those steps can be easily reproduced just by checkout this branch.

[Here](https://github.com/kyma-incubator/service-catalog/blob/94576d1cb8b0f89fd4a775c0810068f43b89affe/contrib/hack/crd/README.md) is the Cookbook for CRDs which provides scripts which you can use to easily bootstrap the minikube with the Service Catalog for the testing purpose.

---
During the testing phase, such issues were found on the master branch:
 - https://github.com/kubernetes-incubator/service-catalog/pull/2629
 - https://github.com/kubernetes-incubator/service-catalog/pull/2628
 - #2627 

### Overview of changes

The overview can be found in [this](https://github.com/kyma-incubator/service-catalog/blob/94576d1cb8b0f89fd4a775c0810068f43b89affe/docs/proposals/crd-implementation.md) doc.

_During the KubeCon Barcelona we can do the f2f introduction (or even a deep dive session) about the CRD implementation._

 
### Actions required to going towards with CRDs
  - Merge tests refactor into master, see PR: https://github.com/kubernetes-incubator/service-catalog/pull/2596
  - Agree about the support for the Aggregated API Server. [Here](https://github.com/kyma-incubator/service-catalog/blob/94576d1cb8b0f89fd4a775c0810068f43b89affe/docs/proposals/going-towards-to-crd.md) you can find document about our idea how to handle that.
  - The migration from the Aggregated API Server to CRD approach is doable and well-documented, see issue: https://github.com/kyma-project/kyma/issues/3779
  - The Service Catalog documentation is updated: https://svc-cat.io/docs/

### Which issues this PR solves:

- https://github.com/kubernetes-incubator/service-catalog/issues/1087
  > (..)we'll need to block the health check until the resources have registered and are gettable.
  
  The common way for CRD healthcheck was implemented and used both in controller and webhook servers, logs:
  ```
  I0519 07:26:10.127589       1 readiness_probe.go:102] CRD "clusterservicebrokers.servicecatalog.k8s.io" is ready
  I0519 07:26:10.127698       1 readiness_probe.go:102] CRD "clusterserviceclasses.servicecatalog.k8s.io" is ready
  I0519 07:26:10.128033       1 readiness_probe.go:102] CRD "clusterserviceplans.servicecatalog.k8s.io" is ready
  I0519 07:26:10.128089       1 readiness_probe.go:102] CRD "servicebindings.servicecatalog.k8s.io" is ready
  I0519 07:26:10.128125       1 readiness_probe.go:102] CRD "servicebrokers.servicecatalog.k8s.io" is ready
  I0519 07:26:10.128142       1 readiness_probe.go:102] CRD "serviceclasses.servicecatalog.k8s.io" is ready
  I0519 07:26:10.128163       1 readiness_probe.go:102] CRD "serviceinstances.servicecatalog.k8s.io" is ready
  I0519 07:26:10.128178       1 readiness_probe.go:102] CRD "serviceplans.servicecatalog.k8s.io" is ready
  I0519 07:26:10.128212       1 readiness_probe.go:111] Readiness probe ready-CRDs checked. There are 8 CRDs
  ```
- https://github.com/kubernetes-incubator/service-catalog/issues/1088
- https://github.com/kubernetes-incubator/service-catalog/issues/2191
  > (..) migrate Service Catalog from using an Aggregated API server toward using CRDs. We should do a prototype of this transition and review it with the larger SIG group.

  CRD approach is implemented and ready for review.

### CODE REVIEW

**Comment by polskikiel** [comment]
I signed it

**Comment by polskikiel** [comment]
singed

**Comment by adamwalach** [comment]
I signed it

**Comment by jberkhahn** [comment]
/lgtm
this is just a test for the new repo

**Comment by mszostok** [comment]
it's ok to review, but I've added hold to ensure that this pr will not be merged by mistake

/hold

**Comment by k** [comment]
Is anyone actively working on this?

**Comment by mszostok** [comment]
@k yep, we still working on that. 

For next week we defined the Epic for finishing the CRD story: https://github.com/kyma-project/kyma/issues/4391

In *AC* section you can see tasks which we created and which will be implemented.

We hope that we will be able to finish it in the next 2 weeks

File: charts/catalog/values.yaml [hunk 1]
```diff
@@ -1,52 +1,25 @@
 # Default values for Service Catalog
 # service-catalog image to use
-image: quay.io/kubernetes-service-catalog/service-catalog:v0.2.1
+image: eu.gcr.io/kyma-project/develop/service-catalog/service-catalog-amd64:crd-0.3.3
```
[the remark this answers is not in the document]
Comment (mszostok) [reply charts/catalog/values.yaml]: definitely, but right now it's helpful to install the chart with changes from this pr for testing purposes

File: charts/catalog/README.md [hunk 2]
```diff
@@ -40,38 +40,18 @@ chart and their default values.
 
 | Parameter | Description | Default |
 |-----------|-------------|---------|
-| `image` | apiserver image to use | `quay.io/kubernetes-service-catalog/service-catalog:v0.2.1` |
+| `image` | Service catalog image to use | `quay.io/kubernetes-service-catalog/service-catalog:v0.1.41` |
```
Comment (mszostok) [inline charts/catalog/README.md]: should be updated

**Comment by mszostok** [comment]
@jberkhahn PTAL I've applied changes after review:
- removed reference to Kyma project
- rebased with current master
- restored the SC image in chart
- moved CRDs definition to a dedicated folder `crds`
- removed the `contrib/hack/crd folder`

**Comment by jberkhahn** [comment]
So, this looks good to me, but I think we need to figure out our Helm chart situation before we merge this.

**Pull request merged**